src/tools/debugger/WebDebugger.java

```diff
@@ -132,4 +134,8 @@ public void startServer(final Debugger dbg) {
     breakpoints = new Breakpoints(dbg);
     connector = new FrontendConnector(breakpoints, instrumenter, this);
   }
+
+  public Map<Source, Set<RootNode>> getRootNodes() {
```

Comment:
I don't like all these getters. If you need only something specific, then add the query method here directly. It looks to me that you want to get the root nodes for a specific source, based on the source URI, and not all of them.